test/distributed/fsdp/test_fsdp_checkpoint.py

```diff
@@ -277,6 +279,116 @@ def test_basic_checkpoint_end_to_end(
         dist.barrier()
 
 
+class CheckpointModule(nn.Module):
+    def __init__(self, checkpoint: bool = False, use_reentrant: bool = True):
+        super().__init__()
+        self.seq = nn.Sequential(*[nn.Linear(100, 100) for _ in range(4)])
+        self.checkpoint = checkpoint
+        self.use_reentrant = use_reentrant
+
+    def forward(self, x):
+        return (
+            checkpoint(self.seq, x, use_reentrant=self.use_reentrant)
+            if self.checkpoint
+            else self.seq(x)
+        )
+
+
+class ModelWithCheckpointSubmodule(nn.Module):
+    def __init__(self, checkpoint: bool = False, use_reentrant: bool = True):
+        super().__init__()
+        self.l1 = nn.Linear(100, 100)
+        self.s1 = CheckpointModule(checkpoint, use_reentrant)
+        self.s2 = CheckpointModule(checkpoint, use_reentrant)
+        self.l2 = nn.Linear(100, 100)
+
+    def forward(self, x):
+        return self.l2(self.s2(self.s1(self.l1(x))))
+
+
+class TestModel(nn.Module):
+    def __init__(self, checkpoint: bool = False, use_reentrant: bool = True):
+        super().__init__()
+        self.l1 = nn.Linear(100, 100)
+        self.m1 = ModelWithCheckpointSubmodule(checkpoint, use_reentrant)
+        self.m2 = ModelWithCheckpointSubmodule(checkpoint, use_reentrant)
```

Comment:
nit: instead of names `m1` and `m2`, checkpoint_1 and checkpoint_2 might make things clearer for readers.

Reply:
sure, let me update that